src/pages/FrontPage.js

```diff
@@ -1,11 +1,214 @@
 import React from 'react';
 import CourseList from '../components/CourseList';
+import LessonFilter from '../components/filter/LessonFilter';
+
+const iconContext = require.context('lessonSrc/', true, /^\.\/[^\/]*\/logo-black\.png/);
+const lessonContext = require.context('onlyFrontmatter!lessonSrc/', true,
+  /^\.\/[^\/]*\/[^\/]*\/(?!README\.md$)[^\/]*\.md/);
 
 const FrontPage = React.createClass({
```

Comment:
Jeg må innrømme at, etter å ha sett på dette i noen timer nå, ennå ikke helt har oversikt over detaljene i hvordan disse tag-strukturene henger sammen. Jeg har på følelsen av at en del av denne koden kan forenkles, men da må jeg forstå litt bedre hva du har tenkt. Kunne du skrevet en litt mer utførlig kommentar (i koden) som forklarer dette i litt mer detalj? Det gjør det også enklere for folk som skal vedlikeholde koden senere. (Alternativt, hvis du klarer å forenkle koden så mye at det er opplagt hva koden gjør, og/eller legger ved nok unit-tests, så kan du krympe inn kommentarene.)